src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -843,3 +1238,59 @@ func (r *runResult) getOrAddIndexSegment(
 		ns.Options().IndexOptions(), ropts)
 	return indexBlockSegment, err
 }
+
+func (r *runResult) mergedResult(other *runResult) *runResult {
+	return &runResult{
+		data:  result.MergedDataBootstrapResult(r.data, other.data),
+		index: result.MergedIndexBootstrapResult(r.index, other.index),
+	}
+}
+
+type shardTimeRangesTimeWindowGroup struct {
+	ranges result.ShardTimeRanges
+	window xtime.Range
+}
+
+func newShardTimeRangesTimeWindowGroups(
+	shardTimeRanges result.ShardTimeRanges,
+	windowSize time.Duration,
+) []shardTimeRangesTimeWindowGroup {
+	min, max := shardTimeRanges.MinMax()
+	estimate := int(math.Ceil(float64(max.Sub(min)) / float64(windowSize)))
+	grouped := make([]shardTimeRangesTimeWindowGroup, 0, estimate)
+	for t := min.Truncate(windowSize); t.Before(max); t = t.Add(windowSize) {
```

Comment:
just making sure: are `shardTimeRanges` exclusive on the max end?

Reply:
They are yup 👍